ee/clickhouse/queries/retention/retention_event_query.py

```diff
@@ -115,7 +115,10 @@ def get_timestamp_field(self) -> str:
             return f"{self.EVENT_TABLE_ALIAS}.timestamp AS event_date"
 
     def _determine_should_join_distinct_ids(self) -> None:
-        self._should_join_distinct_ids = True
+        if self._filter.aggregation_group_type_index is not None:
```

Comment:
Unrelated optimization right?

Reply:
don't want to join unnecessarily. if person props are needed in group aggregation, the initialization should still set this correctly